### PR TITLE
Update Azure.Core to 1.17.0

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableClientOptions.cs [hunk 1]
```diff
@@ -16,7 +16,6 @@ public class TableClientOptions : ClientOptions
         /// library.
         /// </summary>
         private const ServiceVersion Latest = ServiceVersion.V2019_02_02;
-        internal static TableClientOptions Default { get; } = new TableClientOptions();
```
Comment (JoshLove-msft) [inline sdk/tables/Azure.Data.Tables/src/TableClientOptions.cs:19]: Is this intentional?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableClientOptions.cs (hunk 1 above)
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/src/TableClientOptions.cs:19]: Yep, conflicts with the new Core property and is not used.

**Comment by pakrym** [comment]
Identity failure is not caused by this PR.

**Pull request merged**